web/thesauruses/haskell/lists.json

```diff
@@ -0,0 +1,386 @@
+      "not-implemented": true
+    },
+    "create_a_mutable_set": {
+      "not-implemented": true
+    },
+    "get_key_from_mutable_set": {
+      "not-implemented": true
+    },
+    "get_value_from_mutable_set": {
+      "not-implemented": true
+    },
+    "get_all_keys_from_mutable_set": {
+      "not-implemented": true
+    },
+    "get_all_values_from_mutable_set": {
+      "not-implemented": true
+    },
+    "swap_key_and_value_in_mutable_set": {
+      "not-implemented": true
+    },
+    "delete_mutable_set": {
+      "not-implemented": true
+    },
+    "create_a_immutable_set": {
+      "code": "import qualified Data.Map as Map\nMap.empty\nMap.singleton key value -- Map containing a single entry\nMap.fromSet generateValueFromKey set\nMap.fromList [(key1, value1), (key2, value2)]",
+      "comment": "choose `Data.Map.Strict` or `Data.Map.Lazy` depending on your requirements"
+    },
+    "get_key_from_immutable_set": {
+      "not-implemented": true
+    },
+    "get_value_from_immutable_set": {
+      "code": "map ! key"
+    },
+    "get_all_keys_from_immutable_set": {
+      "code": "keys map"
+    },
+    "get_all_values_from_immutable_set": {
+      "code": "elems map"
+    },
+    "swap_key_and_value_in_immutable_set": {
+      "not-implemented": true
+    },
+    "delete_immutable_set": {
+      "not-implemented": true
+    },
+    "find_element_at_position": {
+    },
+    "find_element_by_value": {
+      "code": "elemIndex elementToSearchFor list"
+    },
+    "find_minimum_element": {
+      "code": "minimum list",
+      "comment": "minimum works on all `Foldables` not just lists"
+    },
+    "find_maximum_element": {
+      "code": "maximum list",
+      "comment": "maximum works on all `Foldables` not just lists"
+    },
+    "convert_list_to_string": {
+      "not-implemented": true,
+      "comment": "A `String` is just a `List` of `Char`s"
+    },
+    "concatenate_two_lists": {
+      "name": "Concatenate two lists together",
+      "code": "list1 ++ list2"
+    },
+    "split_list_at_index": {
+      "name": "Split lists at an index",
+      "code": "splitAt index listToSplit"
+    },
+    "split_list_at_value": {
+      "not-implemented": true
+    },
+    "duplicate_a_list": {
+      "not-implemented": true,
+      "comment": "Since you cannot mutate a list it is always going to be the same and you can refer to the original list"
+    },
+    "duplicate_subset_of_list": {
+      "not-implemented": true
+    },
+    "get_list_length": {
+      "code": "length list"
+    },
+    "resize_list": {
+      "comment": "The list is always gonna be exactly as big as the number of elements that are in it",
+      "not-implemented": true
+    },
+    "do_two_lists_match_exactly": {
+      "code": "value = listA == listB"
+    },
+    "do_two_lists_contain_same_items": {
+      "code": "value = listA == listB"
+    },
+    "does_list_satisfy_some_expression": {},
+    "does_list_not_satisfy_an_expression": {},
```

Comment:
I'm not sure what these are supposed to be.

Reply:
`does_list_satisfy_some_expression` and `does_list_not_satisfy_an_expression` are kind of like "all", "some", or "none" across a list. Like "are all elements in this list >5?" or "do some of the items not contain a certain word"? Some languages have functions like these, some don't.